### PR TITLE
feat(blog): Add React Query post.

### PR DESCRIPTION
**Pull request by thorwebdev**
FYI @Gbuomprisco @psteinroe 

https://zone-www-dot-com-git-thor-blog-react-query-ssr-supabase.vercel.app/blog/react-query-nextjs-app-router-cache-helpers

@RamiroND fyi, still waiting on proper og image 👍

TODO:

- [x] Update OG image.

### CODE REVIEW

File: apps/www/_blog/2024-01-12-react-query-nextjs-app-router-cache-helpers.mdx [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+
+```bash
+npm install @supabase/supabase-js @tanstack/react-query @supabase/ssr @supabase-cache-helpers/postgrest-react-query
+```
+
+### Creating a React Query client
+
+Create a React Query client in the root of your component tree. In Next.js app router applications, this is the `layout.tsx` file in the `app` folder.
+
+The `QueryClientProvider` can only be used in client components and can't be directly embeeded in the `layout.tsx` file. Therefore make sure to create a client component first, e.g.
```
Comment (psteinroe) [inline apps/www/_blog/2024-01-12-react-query-nextjs-app-router-cache-helpers.mdx:59]:
```suggestion
The `QueryClientProvider` can only be used in client components and can't be directly embedded in the `layout.tsx` file. Therefore make sure to create a client component first, e.g.
```

File: apps/www/_blog/2024-01-12-react-query-nextjs-app-router-cache-helpers.mdx [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+            staleTime: 60 * 1000,
+          },
+        },
+      })
+  )
+  return <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+}
+```
+
+Next, wrap the root in `laout.tsx`:
```
Comment (psteinroe) [inline apps/www/_blog/2024-01-12-react-query-nextjs-app-router-cache-helpers.mdx:84]:
```suggestion
Next, wrap the root in `layout.tsx`:
```

File: apps/www/_blog/2024-01-12-react-query-nextjs-app-router-cache-helpers.mdx [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+### Creating supabase-js clients for client and server components
+
+To help you utilize the full power of supabase-js, including [Supabase Auth](/auth) and [Row Level Security (RLS)](/docs/guides/auth/row-level-security) policies, we provide the [Supabase SSR helper library](/docs/guides/auth/server-side/creating-a-client) that allows you to conveniently create both browser Supabase clients for client components and server Supabase clients for server components.
+
+Further reading: [detailed documentation for Supabase SSR in Next.js](/docs/guides/auth/server-side/creating-a-client?framework=nextjs)
+
+#### Create a TypedSupabaseClient type
+
+To make sure we have the proper typing avaialable in all our components, we can create a `TypedSupabaseClient` type that we can hand to React Query:
```
Comment (psteinroe) [inline apps/www/_blog/2024-01-12-react-query-nextjs-app-router-cache-helpers.mdx:147]:
```suggestion
To make sure we have the proper typing available in all our components, we can create a `TypedSupabaseClient` type that we can hand to React Query:
```

File: apps/www/_blog/2024-01-12-react-query-nextjs-app-router-cache-helpers.mdx [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+      name
+    `
+    )
+    .eq('id', countryId)
+    .throwOnError()
+    .single()
+}
+```
+
+This is a simple query functions that takes in either the browser or the server Supabase client and the id of a country, and returns a supabase-js query.
```
Comment (psteinroe) [inline apps/www/_blog/2024-01-12-react-query-nextjs-app-router-cache-helpers.mdx:238]:
```suggestion
This is a simple query function that takes in either the browser or the server Supabase client and the id of a country, and returns a supabase-js query.
```

**Review by psteinroe — APPROVED** [review]
LGTM!

Just found a few typos

**Review by RamiroND — APPROVED** [review]

**Pull request merged**